test/example/SnapshotGateTest.js

```diff
@@ -0,0 +1,717 @@
+          await snapshotGate.connect(deployer).appendToSnapshot(snapshot);
+
+          // Freeze the snapshot
+          await snapshotGate.connect(deployer).freezeSnapshot();
+
+          // Grab an entry from the snapshot
+          let entry = snapshot[Math.floor(snapshot.length / 4)];
+
+          // Offer, token, and group ids are aligned for the sake of sanity,
+          offerId = entry.tokenId;
+
+          // Get the account to make the call with
+          let holder = holderByAddress[entry.owner];
+
+          // Exhaust all commits for this buyer for this token
+          for (let i = 0; i < entry.amount; i++) {
+            await snapshotGate
+              .connect(holder)
+              .commitToGatedOffer(entry.owner, offerId, entry.tokenId, { value: price });
+          }
+
+          // Commit to the offer
+          await expect(
+            snapshotGate.connect(holder).commitToGatedOffer(entry.owner, offerId, entry.tokenId, { value: price })
+          ).to.revertedWith("Buyer's balance of the snapshot token id has been used");
+        });
+
+        it("condition specifies a different tokenId from the one given", async function () {
+          // Upload the snapshot
+          await snapshotGate.connect(deployer).appendToSnapshot(snapshot);
+
+          // Freeze the snapshot
+          await snapshotGate.connect(deployer).freezeSnapshot();
+
+          // Grab first entry from the snapshot
+          let entry = snapshot[0];
+
+          // Grab second entry from the snapshot
+          let entry2 = snapshot[1];
+
+          // wrong offer id
+          offerId = entry2.tokenId;
+
+          // Get the account to make the call with
+          let holder = holderByAddress[entry.owner];
+
+          // Commit to the offer
+          await expect(
+            snapshotGate.connect(holder).commitToGatedOffer(entry.owner, offerId, entry.tokenId, { value: price })
+          ).to.revertedWith("Condition specifies a different tokenId from the one given");
+        });
+
+        it("offer is from another seller", async function () {
+          // Upload the snapshot but don't freeze
+          await snapshotGate.connect(deployer).appendToSnapshot(snapshot);
+
+          // Freeze the snapshot
+          await snapshotGate.connect(deployer).freezeSnapshot();
+
+          // Grab an entry from the snapshot
+          let entry = snapshot[Math.floor(snapshot.length / 4)];
+
+          // Offer, token, and group ids are aligned for the sake of sanity
+          offerId = entry.tokenId;
+
+          // Get the account to make the call with
+          let caller = holderByAddress[entry.owner];
+
+          // Commit to the offer
+          await expect(
+            snapshotGate
+              .connect(caller)
+              .commitToGatedOffer(caller.address, otherSellerOfferId, entry.tokenId, { value: price })
+          ).to.revertedWith("Offer is from another seller");
+        });
+      });
+    });
```

Comment:
Need test cases for insufficient payment (which I think should be "Incorrect payment amount" -- see above) and transfer to gate not approved.